ui/app/AppLayouts/Chat/views/ChatMessagesView.qml

```diff
@@ -42,6 +43,20 @@ Item {
     signal openStickerPackPopup(string stickerPackId)
     signal showReplyArea(string messageId, string author)
 
+    property bool allMessagesLoaded: false
+
+    onIsActiveChannelChanged: {
+        if (!isActiveChannel) {
+            root.allMessagesLoaded = false
+            return
+        }
+        // We wait to load all messages, because switching back to chats makes
+        // the scroll go crazy so it loads way too many messages making it slow
+        timer.setTimeout(function() {
```

Comment:
Maybe I miss things, but this looks like a workaround. The data model should handle the async loading and the asynchronous process be transparent to the View. @jrainville, how about a follow-up task to refactor this, or do you think it is final?

Reply:
> The data model should handle the async loading and the asynchronous process be transparent to the View

Yeah, I know, but somehow, the ListView was loading way too many items because, for reasons I don't know, it was moving the scroll when switching. This made it so that lots of rendering was done for no reason.

> how about a follow-up task to refactor this, or do you think it is final?

This shouldn't be final, but I couldn't find another solution as of now. Maybe when we do refactor the QML, we'll be rid of the problem or find a cleaner solution.